src/eradiate/experiments/_core.py

```diff
@@ -478,11 +479,30 @@ def process(
 
         for ctx_index, spectral_group_dict in mi_results.items():
             for sensor_id, mi_bitmap in spectral_group_dict.items():
-                measure = sensor_to_measure[sensor_id]
-                measure.mi_results[ctx_index] = {
-                    "bitmap": mi_bitmap,
-                    "spp": spp if spp > 0 else measure.spp,
-                }
+                
+                if self.integrator.moment :
+                    split = mi_bitmap.split()
+                    
+                    m2 = split[1][1]
+
+                    # convert to single channel
+                    if m2.channel_count() > 1:
+                        m2_np = np.array(m2, copy=False)[:,:,0:1] # note the slice is to keep the channel dimension
```

Comment:
I'd rather make this explicit, either with
```suggestion
                        m2_np = np.array(m2, copy=False)[:,:,[0]] # note the slice is to keep the channel dimension
```
or
```suggestion
                        m2_np = np.atleast_3d(np.array(m2, copy=False)[..., 0])
```